File: Vowel Balance/Vowel Balance.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int is_vowel(char c) {
    c = tolower(c);
    return (c=='a' || c=='e' || c=='i' || c=='o' || c=='u');
}

int is_balanced(const char *s) {
    int n = strlen(s);
    int half = n / 2;
    int countFirst = 0, countSecond = 0;

    for (int i = 0; i < half; i++) {
        if (is_vowel(s[i])) countFirst++;
    }
    for (int i = n - half; i < n; i++) {
        if (is_vowel(s[i])) countSecond++;
    }

    return countFirst == countSecond;
}

int main() {
    printf("%d\n", is_balanced("racecar")); // 1
    printf("%d\n", is_balanced("Lorem Ipsum")); // 1
    printf("%d\n", is_balanced("Kitty Ipsum")); // 0
    return 0;
}

File: Vowel Balance/Vowel Balance.js
function isBalanced(s) {
  const vowels = "aeiouAEIOU";
  const n = s.length;
  const half = Math.floor(n / 2);

  const first = s.slice(0, half);
  const second = s.slice(n - half);

  const count = str => [...str].filter(c => vowels.includes(c)).length;

  return count(first) === count(second);
}

console.log(isBalanced("racecar")); // true
console.log(isBalanced("Lorem Ipsum")); // true
console.log(isBalanced("Kitty Ipsum")); // false
